### PR TITLE
[QUILL-1542]/Jia/action-sheet

### PR DESCRIPTION
**Pull request by jia-deriv**
Props:
`ActionSheet.Root`
A provider to store value using in the action sheet.
- `isOpen`
  - the value for setting action sheet open or close
- `position`
  - position left or right action sheet for side sheet
- `type`
  - modal or non-modal action sheet
- `expandable`
  - action sheet is draggable
- `onOpen`
  - onOpen function
- `onClose`
  - onClose function

`ActionSheet.Trigger`
A component to handle trigger button.
- `iconComponent`
  - a props for trigger icon if not icon can use children
- `iconClassName`
  - classname for icon trigger
- `iconSize`
  - props for icon size

`ActionSheet.Portal`
A component to handle expandable drag. Props same as a div component

`ActionSheet.Content`
Body of the action sheet. Props same as a div component

`ActionSheet.Footer`
Footer part of the action sheet
- `primaryAction`
  - `onAction` : function
  - `content` : button label
- `secondaryActionHandler`
  - `onAction` : function
  - `content` : button label
- `alignment`
  - to set button horizontal or vertical
  
Example usage:
```
<ActionSheet.Root {...props} isOpen={open} onOpen={openHandler}>
    <ActionSheet.Trigger>Click Here</ActionSheet.Trigger>
    <ActionSheet.Portal>
        <ActionSheet.Header
            title={title}
            description={description}
            closeIcon={closeIcon}
            icon={icon}
        />
        <ActionSheet.Content className="mock-action-sheet--content">
            <Text size="sm">
                Bottom sheet is a surface fixed at the bottom of the
                screen which includes content related to the
                previous screen.
            </Text>
            <Text size="sm">
                Bottom sheet is a surface fixed at the bottom of the
                screen which includes content related to the
                previous screen.
            </Text>
            <Text size="sm">
                Bottom sheet is a surface fixed at the bottom of the
                screen which includes content related to the
                previous screen.
            </Text>
        </ActionSheet.Content>
        <ActionSheet.Footer
            primaryAction={primaryAction}
            secondaryAction={secondaryAction}
            alignment={alignment}
        />
    </ActionSheet.Portal>
</ActionSheet.Root>

```

### CODE REVIEW

File: lib/components/ActionSheet/footer/index.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import { FooterProps } from "../types";
+// import Button from "../../button/basic";
```
Comment (prince-deriv) [inline lib/components/ActionSheet/footer/index.tsx:2]: let's remove this

File: lib/components/ActionSheet/footer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import clsx from "clsx";
+import "./footer.scss";
+import { Button } from "../../Button";
+
+const Footer = ({
+    primaryAction,
+    secondaryAction,
+    alignment,
+    className,
+    ...restProps
```
Comment (prince-deriv) [inline lib/components/ActionSheet/footer/index.tsx:14]:
```suggestion
    ...rest
```

**Review by prince-deriv — APPROVED** [review]

File: lib/components/ActionSheet/footer/index.tsx (hunk 2 above)
Comment (jia-deriv) [reply lib/components/ActionSheet/footer/index.tsx]: done

File: lib/components/ActionSheet/footer/index.tsx (hunk 1 above)
Comment (jia-deriv) [reply lib/components/ActionSheet/footer/index.tsx]: done

**Pull request merged**